src/core/init/initialize_media_source.ts

```diff
@@ -200,11 +204,20 @@ export default function InitializeOnMediaSource(
   const mediaError$ = throwOnMediaError(mediaElement);
 
   const loadContent$ = observableCombineLatest([
-    openMediaSource$,
+    openMediaSource$.pipe(startWith(null)),
+    emeManager$.pipe(filter(isEMEReadyEvent),
+                     mapTo(true),
+                     take(1),
+                     startWith(false)),
     fetchManifest(url, undefined),
-    emeManager$.pipe(filter(isEMEReadyEvent), take(1)),
-  ]).pipe(mergeMap(([ initialMediaSource, initialManifestObj ]) => {
-    const { manifest } = initialManifestObj;
+  ]).pipe(mergeMap(([ initialMediaSource, isEMEReady, manifestEvt ]) => {
+    if (manifestEvt.type === "warning") {
```

Comment:
Isn't there a risk that openMediaSource$ and emeManager$ emit twice, and fetchManifest once with warning ? So we return twice the warning ?

Reply:
You're right!

I'm re-working that thing

Reply:
Done!

I've also re-worked the `schedulerWarnings$` subject which by never completing could lead the corresponding manifestFetcher observable to never complete.

Reply:
Wait, no, it's still not good...

Reply:
Done, though it can look a little ugly...
You're the judge.